contracts/manifold/lazyclaim/ERC721LazyPayableClaim.sol

```diff
@@ -95,199 +105,256 @@ contract ERC721LazyPayableClaim is IERC165, IERC721LazyPayableClaim, ICreatorExt
+    function mint(address creatorContractAddress, uint256 instanceId, uint32 mintIndex, bytes32[] calldata merkleProof, address mintFor) external payable override {
+        Claim storage claim = _getClaim(creatorContractAddress, instanceId);
 
         // Check totalMax
-        require(++claim.total <= claim.totalMax || claim.totalMax == 0, "Maximum tokens already minted for this claim");
+        require((++claim.total <= claim.totalMax || claim.totalMax == 0) && claim.total <= MAX_UINT_24, "Maximum tokens already minted for this claim");
 
         // Validate mint
-        _validateMint(creatorContractAddress, claimIndex, claim.startDate, claim.endDate, claim.walletMax, claim.merkleRoot, mintIndex, merkleProof, mintFor);
+        _validateMint(creatorContractAddress, instanceId, claim.startDate, claim.endDate, claim.walletMax, claim.merkleRoot, mintIndex, merkleProof, mintFor);
 
         // Transfer funds
         _transferFunds(claim.erc20, claim.cost, claim.paymentReceiver, 1, claim.merkleRoot != "");
 
         // Do mint
-        uint256 newTokenId = IERC721CreatorCore(creatorContractAddress).mintExtension(msg.sender);
-
-        // Insert the new tokenId into _tokenClaims for the current claim address & index
-        _tokenClaims[creatorContractAddress][newTokenId] = TokenClaim(uint224(claimIndex), claim.total);
+        if (claim.contractVersion >= 3) {
+            uint80 tokenData = uint56(instanceId) << 24 | uint24(claim.total);
+            IERC721CreatorCore(creatorContractAddress).mintExtension(msg.sender, tokenData);
+        } else {
+            uint256 newTokenId = IERC721CreatorCore(creatorContractAddress).mintExtension(msg.sender);
+            // Insert the new tokenId into _tokenClaims for the current claim address & instanceId
+            _tokenClaims[creatorContractAddress][newTokenId] = TokenClaim(uint224(instanceId), claim.total);
+        }
 
-        emit ClaimMint(creatorContractAddress, claimIndex);
+        emit ClaimMint(creatorContractAddress, instanceId);
     }
 
     /**
      * See {ILazyPayableClaim-mintBatch}.
      */
-    function mintBatch(address creatorContractAddress, uint256 claimIndex, uint16 mintCount, uint32[] calldata mintIndices, bytes32[][] calldata merkleProofs, address mintFor) external payable override {
-        Claim storage claim = _getClaim(creatorContractAddress, claimIndex);
+    function mintBatch(address creatorContractAddress, uint256 instanceId, uint16 mintCount, uint32[] calldata mintIndices, bytes32[][] calldata merkleProofs, address mintFor) external payable override {
+        Claim storage claim = _getClaim(creatorContractAddress, instanceId);
 
         // Check totalMax
         claim.total += mintCount;
-        require(claim.totalMax == 0 || claim.total <= claim.totalMax, "Too many requested for this claim");
+        require((claim.totalMax == 0 || claim.total <= claim.totalMax) && claim.total <= MAX_UINT_24, "Too many requested for this claim");
```

Comment:
claim.total is uint32, why 24?

Reply:
To fit it in the packed struct nonce (the new creator opttimization).